zero_sum_eval/games/chess/chess_dataset.py

```diff
@@ -10,10 +10,11 @@
 
 @DATASET_REGISTRY.register("chess_dataset")
 class ChessDataset(Dataset):
-    def __init__(self, filename: str, role: Union[Literal["White"], Literal["Black"]]) -> None:
+    def __init__(self, filename: str, role: Union[Literal["White"], Literal["Black"]], num_samples: Optional[int] = None) -> None:
```

Comment:
can we pick one of samples/examples and use it consistently? I think samples may be better

Reply:
Anyways, it seems the changes in this file are unrelated.

Reply:
Examples is the term used in DSPy so I guess it makes more sense as you've done in the new commit.